esprima.js

```diff
@@ -3955,6 +3956,12 @@
             }
         }
 
+        if (matchKeyword('let') || matchContextualKeyword('let')) {
```

Comment:
Consider adding a function called `matchLet` as this logic is used in several places.